### PR TITLE
Add the ytt based release pipeline

### PR DESCRIPTION
**Pull request by beeender**
The release pipeline will:
- Automatically make release build for every commit.
- The final release step needs to be started manually.
  - Then the build will be pushed to the release bucket.
  - And a tag will be pushed to the repository.

### CODE REVIEW

**Comment by beeender** [comment]
Pipeline can be seen at

https://extensions.ci.gpdb.pivotal.io/teams/main/pipelines/RELEASE:diskquota

File: concourse/fly.sh [hunk 1]
```diff
@@ -73,6 +73,17 @@ case ${pipeline_config} in
       pipeline_name="DEV:${pipeline_name}"
       config_file="dev.yml"
     ;;
+  release)
+      if [ -z "${pipeline_name}" ]; then
+          pipeline_name="RELEASE:diskquota"
```
Comment (beeender) [inline concourse/fly.sh:78]: Probably we use the name `RELEASE:diskquota:gpdb` is better, to align with the commit pipeline? I think I am going to change this if no objection.

Comment (beeender) [inline concourse/fly.sh:78]: Probably we use the name `RELEASE:diskquota:gpdb` is better, to align with the commit pipeline? I think I am going to change this if no objection.

File: concourse/pipeline/trigger_def.lib.yml [hunk 2]
```diff
@@ -54,14 +58,44 @@ on_error:
...
+auto_trigger: true
+to_get:
+- get: diskquota_src
+  resource: diskquota_commit
+to_put_pre: #@ []
+to_put_post:
+- put: diskquota_commit
+  params:
+    repository: diskquota_src
+    tag: diskquota_src/VERSION
```
Comment (Sasasu) [inline concourse/pipeline/trigger_def.lib.yml:87]: is this a placeholder? I don't find the logic for the version(tag) generate.

**Review by Sasasu — APPROVED** [review]

File: concourse/pipeline/trigger_def.lib.yml (hunk 2 above)
Comment (beeender) [reply concourse/pipeline/trigger_def.lib.yml:87]: It is black magic from concourse git resource 

> tag: Optional. If this is set then HEAD will be tagged. The value should be a path to a file containing the name of the tag.

https://github.com/concourse/git-resource

**Review by xuebinsu — APPROVED** [review]

**Pull request merged**